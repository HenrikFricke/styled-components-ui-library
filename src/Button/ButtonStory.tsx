import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { PrimaryButton, SecondaryButton } from './Button';

storiesOf('Button', module)
  .add('primary', () => (
    <PrimaryButton onClick={() => alert('toll')}>I'am a button</PrimaryButton>
  ))
  .add('secondary', () => (
    <SecondaryButton>I'am a button</SecondaryButton>
  ));