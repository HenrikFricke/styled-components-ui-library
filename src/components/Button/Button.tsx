import styled from 'styled-components';

import {
    colors,
    transitions,
    fonts
} from '../../constants';

export const PrimaryButton = styled.button`
    border: 2px solid ${colors.PRIMARY};
    background-color: transparent;
    color: ${colors.PRIMARY};
    padding: 10px;
    border-radius: 4px;
    cursor: pointer;
    transition: ${transitions.STANDARD} all;
    font-size: ${fonts.STANDARD};

    &:hover {
        background-color: ${colors.PRIMARY};
        color: ${colors.SECONDARY};
    }
`;

export const SecondaryButton = PrimaryButton.extend`
    background-color: ${colors.PRIMARY};
    color: ${colors.SECONDARY};

    &:hover {
        background-color: transparent;
        color: ${colors.PRIMARY};
    }
`;