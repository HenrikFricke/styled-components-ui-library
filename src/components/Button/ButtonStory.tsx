import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { PrimaryButton, SecondaryButton } from './Button';

storiesOf('Button', module)
  .add('primary', () => (
    <PrimaryButton>I'am a button</PrimaryButton>
  ))
  .add('secondary', () => (
    <SecondaryButton>I'am a button</SecondaryButton>
  ));