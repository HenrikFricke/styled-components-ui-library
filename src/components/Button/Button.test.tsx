import * as React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import 'jest-styled-components';

import { PrimaryButton, SecondaryButton } from './Button';

describe('PrimaryButton', () => {
    let tree: ReactTestRenderer.ReactTestRendererJSON;

    beforeEach(() => {
        tree = ReactTestRenderer.create(
            <PrimaryButton>I'am a button</PrimaryButton>
        ).toJSON();
    });

    it('should render correctly', () => {
        expect(tree).toMatchStyledComponentsSnapshot();
    });
});

describe('SecondaryButton', () => {
    let tree: ReactTestRenderer.ReactTestRendererJSON;

    beforeEach(() => {
        tree = ReactTestRenderer.create(
            <SecondaryButton>I'am a button</SecondaryButton>
        ).toJSON();
    });

    it('should render correctly', () => {
        expect(tree).toMatchStyledComponentsSnapshot();
    });
});
