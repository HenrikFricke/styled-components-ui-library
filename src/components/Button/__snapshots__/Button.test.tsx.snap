// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton should render correctly 1`] = `
.gqsqPu {
  border: 2px solid palevioletred;
  background-color: palevioletred;
  color: white;
  padding: 10px;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 300ms all;
  transition: 300ms all;
  font-size: 16px;
}

.gqsqPu:hover {
  background-color: mediumvioletred;
  border-color: mediumvioletred;
}

<button
  className="sc-bdVaJa gqsqPu"
>
  I'am a button
</button>
`;

exports[`SecondaryButton should render correctly 1`] = `
.jujSlz {
  border: 2px solid palevioletred;
  background-color: palevioletred;
  color: white;
  padding: 10px;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 300ms all;
  transition: 300ms all;
  font-size: 16px;
  background-color: transparent;
  color: palevioletred;
}

.jujSlz:hover {
  background-color: mediumvioletred;
  border-color: mediumvioletred;
}

.jujSlz:hover {
  background-color: transparent;
  color: mediumvioletred;
}

<button
  className="sc-bwzfXH jujSlz"
>
  I'am a button
</button>
`;
